ext/storage_sources/s3_store/s3_connection.h

```diff
@@ -15,14 +15,15 @@
 class S3Connection {
     public:
     explicit S3Connection(const Aws::S3Crt::ClientConfiguration &config);
-    bool ListBuckets(std::vector<std::string> &buckets) const;
-    bool ListObjects(const std::string &bucketName, std::vector<std::string> &objects) const;
-    bool PutObject(const std::string &bucketName, const std::string &objectKey,
+    int ListBuckets(std::vector<std::string> &buckets) const;
+    int ListObjects(const std::string &bucketName, const std::string &prefix,
+      std::vector<std::string> &objects, uint32_t batchSize = 1000, bool listSingle = false) const;
+    int PutObject(const std::string &bucketName, const std::string &objectKey,
       const std::string &fileName) const;
-    bool DeleteObject(const std::string &bucketName, const std::string &objectKey) const;
+    int DeleteObject(const std::string &bucketName, const std::string &objectKey) const;
     ~S3Connection() = default;
 
     private:
-    const Aws::S3Crt::S3CrtClient m_S3CrtClient;
+    const Aws::S3Crt::S3CrtClient m_s3CrtClient;
```

Comment:
We don't use the `m_` prefix for class members. The variable name should be `s3CrtClient`.